### ISSUE TITLE
[libbacktrace] Empty backtrace

### ISSUE THREAD

**Issue by kmeher** [msys2/MINGW-packages#20020]
### Description / Steps to reproduce the issue

Environment : UCRT64 (Windows 11)
1 - Create a test code :
```
#include <backtrace.h>
#include <cstdio>
#include <iostream>
#include <windows.h>
#include <psapi.h>
#include <iomanip>

static struct backtrace_state * bstate = nullptr;

void hello1()
{
	backtrace_print(bstate, 0, stdout);
}

void hello2()
{
	hello1();
}

static void state_err_cb(void *data, const char *msg, int errnum)
{
	std::cout << "backtrace_create_state error :" << std::endl
			<< "\tmsg = " << msg << std::endl
			<< "\terrnum = " << errnum << std::endl;
}

int main()
{
	HANDLE hproc = GetCurrentProcess();
	char proc_file[MAX_PATH] = {0};
	if (GetModuleFileNameExA(hproc, nullptr, proc_file, MAX_PATH) == 0)
	{
		std::cout << "GetModuleFileNameExA error: err = 0x" << std::hex << std::setw(8) << std::setfill('0') << GetLastError() << std::endl;
		return 2;
	}
	std::cout << "proc_file = \"" << proc_file << "\"" << std::endl;
	bstate = backtrace_create_state(proc_file, 0, state_err_cb, nullptr);
	if (bstate == nullptr)
	{
		return 1;
	}
	hello2();
	return 0;
}
```
2 - Compile it (in UCRT64 MSYS2 console)
`g++ -g -o test.exe main.cpp -lbacktrace`
3 - Run it

With the package "ucrt64/mingw-w64-ucrt-x86_64-libbacktrace r53.1da441c-3" the result is not correct (as listed in the actual behavior), while with a compiled GIT clone the results are correct (as listed in the expected behavior).

### Expected behavior

```
proc_file = "D:\Work\src\heppers2\heppers\t\test2.exe"
0xaf06147f _Z6hello1v
        ../tests/main.cpp:12
0xaf061493 _Z6hello2v
        ../tests/main.cpp:17
0xaf0616c7 main
        ../tests/main.cpp:42
0xaf0612ed __tmainCRTStartup
        C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtexe.c:267
0xaf061405 mainCRTStartup
        C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtexe.c:188
0xa376257c ???
        ???:0
0xa4a8aa57 ???
        ???:0
0xffffffff ???
        ???:0
```

### Actual behavior

```
proc_file = "D:\Work\src\heppers2\heppers\t\test.exe"
0x5a68147f ???
        ???:0
0x5a681493 ???
        ???:0
0x5a6816c7 ???
        ???:0
0x5a6812ed ???
        ???:0
0x5a681405 ???
        ???:0
0xa376257c ???
        ???:0
0xa4a8aa57 ???
        ???:0
0xffffffff ???
        ???:0
```

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-22631

### MINGW environments affected

- [ ] MINGW64
- [ ] MINGW32
- [X] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

**Comment by Biswa96** [comment]
Isn't the expected behavior correct? The binaries are stripped before packaging.

**Comment by MehdiChinoune** [comment]
> Isn't the expected behavior correct? The binaries are stripped before packaging.

That's the output of the executable he built against libbacktrace (It is not a package of MSYS2)

**Comment by kmeher** [comment]
> Isn't the expected behavior correct? The binaries are stripped before packaging.

The expected behavior is the correct (expected) behavior which I got by using a compiled version of libbacktrace on my computer and linking to it. What I have right now is the "actual behavior" by linking to the package provided in the UCRT64 repository.

**Comment by Biswa96** [comment]
Updating the package with latest commit fixes the issue.

**Comment by Biswa96** [comment]
Fixed in 1a06ccc25def7192cb74324e81e06da518764002 commit.
